core/src/main/scala/com/nvidia/spark/rapids/tool/profiling/Profiler.scala

```diff
@@ -172,16 +172,29 @@ class Profiler(hadoopConf: Configuration, appArgs: ProfileArgs, enablePB: Boolea
           UnknownAppResult(pathStr, "", errorMessage)
         case Right(app: ApplicationInfo) =>
           // Case with successful creation of ApplicationInfo
+          // If the app has a newer attemptId being processed, skip this attemptId
+          if (app.isOlderAttempt) {
+            throw OlderAttemptException(app)
+          }
           processSuccessApp(app)
+          if (app.isOlderAttempt) {
+            throw OlderAttemptException(app)
+          }
```

Comment:
For profiler, I don't see this code addressing the problem. Profiler writes to the directory once an app is processed.
The simplest solution is to synchronize of the writer object. this way only attempts with higher value will be exclusively writing to disk.
If an awaiting thread has less attempt number, then it throws the exception.